### PR TITLE
added loggers to investigate address validation issue in prod

### PR DESCRIPTION
**Pull request by kritikarai6**

### CODE REVIEW

File: app/models/backend/Address.scala [hunk 1]
```diff
@@ -104,7 +107,17 @@ object Address {
 
   def apply(addressRecord: AddressRecord): Address = {
     val addressLines: Seq[Option[String]] =
-      (addressRecord.address.lines.map(s => Option(s).filter(_.trim.nonEmpty))
+      (addressRecord.address.lines.map { s =>
+        val trimmedLine = s.trim
```
Comment (owen-opencast) [inline app/models/backend/Address.scala:111]: Can you do the check on the untrimmed line? I think the trim is just used to identify whether the line is empty

File: app/connectors/CitizenDetailsConnector.scala [hunk 2]
```diff
@@ -98,6 +98,9 @@ class CitizenDetailsConnector @Inject() (http: HttpClientV2, appConfig: Frontend
                 s"Bad request for postcode ${address.postcode.getOrElse("-")} [${headerCarrier.sessionId.map(_.value).getOrElse("-")}]"
               )
             }
+            logger.warn(
+              s"Address for Bad request for postcode ${address.postcode.getOrElse("-")} [${headerCarrier.sessionId.map(_.value).getOrElse("-")}] [${address.line1}] [${address.line4}]"
```
Comment (owen-opencast) [inline app/connectors/CitizenDetailsConnector.scala:102]: Can we keep the line you've added, but remove the logging I had above, as the new logging is capturing all errors now. Change the message on 102 to something like "Bad request for address", and then the details, as we're not just printing out for bad postcodes.

**Review by owen-opencast — CHANGES_REQUESTED** [review]

File: app/connectors/CitizenDetailsConnector.scala (hunk 2 above)
Comment (kritikarai6) [reply app/connectors/CitizenDetailsConnector.scala]: Done

File: app/models/backend/Address.scala (hunk 1 above)
Comment (kritikarai6) [reply app/models/backend/Address.scala]: Done

**Review by owen-opencast — APPROVED** [review]

**Pull request merged**